src/sage/combinat/partitions.pyx

```diff
@@ -179,3 +335,400 @@ def ZS1_iterator_nk(int n, int k):
+
+    .. SEEALSO::
+
+        :meth:`sage.combinat.partitions.ZS1_iterator`
+
+    EXAMPLES::
+
+        sage: from sage.combinat.partitions import AccelDesc_iterator
+        sage: for p in AccelDesc_iterator(4):
+        ....:     print(p)
+        [4]
+        [3, 1]
+        [2, 2]
+        [2, 1, 1]
+        [1, 1, 1, 1]
+        sage: next(AccelDesc_iterator(4))
+        [4]
+        sage: type(_)
+        <class 'list'>
+    """
+    # Easy cases.
+    if n < 0:
+        return
+    if n <= 1:
+        yield [1] * n
+        return
+    cdef list P = [1] * n
+    P[0] = n
+    yield [n]
+    cdef int m = 0
+    cdef int h = 0
+    while h >= 0:
+        AccelDesc_step(P, n, &m, &h)
+        yield P[:m+1]
+
+
+def AccelDesc_next(list P):
+    r"""
+    Return the partition after `P` in the ordering of the AccelDesc algorithm.
+
+    INPUT:
+
+    - ``P`` -- a list encoding a partition of an integer `n` in descending order
+      (i.e., `P_i \geq P_{i+1}`)
+
+    EXAMPLES::
+
+        sage: from sage.combinat.partitions import AccelDesc_iterator, AccelDesc_next
+        sage: P = [4]
+        sage: while P:
+        ....:     print(P)
+        ....:     P = AccelDesc_next(P)
+        [4]
+        [3, 1]
+        [2, 2]
+        [2, 1, 1]
+        [1, 1, 1, 1]
+        sage: A = [list(p) for p in Partitions(7)]
+        sage: all(AccelDesc_next(p) == q for p, q in zip(A, A[1:]))
+        True
+    """
+    if P[0] == 1:
+        return
+    if P[-1] != 1:
+        return P[:-1] + [P[-1] - 1, 1]
+
+    cdef int n = sum(P)
+    cdef int m = len(P) - 1  # index of the last element in P
+    # Search for the index h of the last value > 1 in P
+    cdef int h = m
+    while P[h] == 1:
+        h -= 1
+    # Let Q be such that Q[:m+1] == P and Q[h+1:] is an array of n-(h+1) ones
+    cdef list Q = P + [1]*(n - m - 1)
+    AccelDesc_step(Q, n, &m, &h)
+    return Q[:m+1]
+
+
+def AccelAsc_iterator(int n):
+    r"""
+    Return an iterator over the partitions of `n`.
+
+    The partitions are generated in the increasing lexicographic order and each
+    partition is represented as a list in ascending order (i.e., `p_i \leq
+    p_{i+1}`).
+
+    This is an implementation of the ``AccelAsc`` algorithm found in [KS2012]_.
+
+    .. SEEALSO::
+
+        :meth:`sage.combinat.partitions.ZS1_iterator`
+
+    EXAMPLES::
+
+        sage: from sage.combinat.partitions import AccelAsc_iterator
+        sage: for p in AccelAsc_iterator(4):
+        ....:     print(p)
+        [1, 1, 1, 1]
+        [1, 1, 2]
+        [1, 3]
+        [2, 2]
+        [4]
+        sage: next(AccelAsc_iterator(4))
+        [1, 1, 1, 1]
+        sage: type(_)
+        <class 'list'>
+    """
+    # Easy cases.
+    if n < 0:
+        return
+    if n <= 1:
+        yield [1] * n
+        return
+    cdef list P = [0] * n
+    cdef int k = 1
+    cdef int y = n - 1
+    cdef int x, ell
+    while k:
+        k -= 1
+        x = P[k] + 1
+        while 2 * x <= y:
```

Comment:
```suggestion
    cdef int x, twox, ell
    while k:
        k -= 1
        x = P[k] + 1
        twox = 2 * x
        while twox <= y:
```
Not sure if this will be faster or not, or if the compiler would optimizes this away anyways.

Reply:
I'm not sure we need this change. My timing don't show any improvement:

Before
```py
sage: %timeit sum(1 for _ in AccelAsc_iterator(20))
140 µs ± 481 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
sage: %timeit sum(1 for _ in AccelAsc_iterator(50))
46.1 ms ± 117 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```
After
```py
sage: %timeit sum(1 for _ in AccelAsc_iterator(20))
141 µs ± 810 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
sage: %timeit sum(1 for _ in AccelAsc_iterator(50))
46.1 ms ± 568 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

Reply:
Either it is such a mirco-optimization as to never matter or the compiler generally optimizes it away. Then let's leave it as is.